web/html/src/manager/admin/config/use-monitoring-api.ts

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import Network, { JsonResult } from "utils/network";
+import { MessageType } from "components/messages";
+
+type ExportersResultType = {
+  exporters: {
+    [key: string]: boolean;
+  };
+  messages: {
+    [key: string]: string;
+  };
+};
+
+function isRestartNeeded(data: ExportersResultType) {
+  return Object.keys(data.exporters).some(key => data.messages[key] === "restart");
+}
+
+const useMonitoringApi = () => {
+  const [action, setAction] = useState<null | "checking" | "enabling" | "disabling">(null);
+  const [exportersStatus, setExportersStatus] = useState<
+    | {
+        [key: string]: boolean;
+      }
+    | null
+    | undefined
+  >(null);
+  const [exportersMessages, setExportersMessages] = useState<{
+    [key: string]: string;
+  }>({});
+  const [restartNeeded, setRestartNeeded] = useState<boolean>(false);
+  const [messages, setMessages] = useState<Array<MessageType>>([]);
+
+  const handleResponseError = (jqXHR: any, arg: string = ""): any => {
```

Comment:
So far we used `jqXHR: any` because we didn't have any proper type definition for it. Is there any type definition in the TS world for `jqXHR`?

Reply:
The jQuery types provide types for this as well, I've made the changes here manually plus added it to the migrator script. The results of our network requests aren't always typed through and through because they're fairly complex promise chains, but we can get to that some other time in the future.